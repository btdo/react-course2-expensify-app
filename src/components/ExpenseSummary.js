import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import totalExpenses from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = ({expenseCount, total }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    return (
    <div> 
        Viewing { expenseCount } {expenseWord} totalling { numeral(total/100).format('$0,0.00') }
    </div>
)};

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        total: totalExpenses(state.expenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);