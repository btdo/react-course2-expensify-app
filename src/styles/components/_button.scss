.button {
    background: $blue;
    border: none;
    color: white;
    padding: $s-size;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    text-decoration: none;
    line-height: 1;
}

.button--secondary {
    background-color: #888888;
}

.button--link{
    background: none;
}