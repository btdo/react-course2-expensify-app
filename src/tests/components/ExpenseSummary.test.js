import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render expense summary with 1 expense', () => {
    const wrapper = shallow (<ExpenseSummary expenseCount={1} total={1223}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should render expense summary with multiple expenses', () => {
    const wrapper = shallow (<ExpenseSummary expenseCount={12} total={121223}/>);
    expect(wrapper).toMatchSnapshot();
})