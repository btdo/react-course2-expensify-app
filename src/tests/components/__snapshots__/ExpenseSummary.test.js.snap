// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $12.23
</div>
`;

exports[`should render expense summary with multiple expenses 1`] = `
<div>
  Viewing 
  12
   
  expenses
   totalling 
  $1,212.23
</div>
`;
